packages/@o3r/core/schematics/rule-factories/cms-adapter/index.ts

```diff
@@ -93,9 +99,16 @@ export function updateCmsAdapter(options: { projectName: string | null }, rootPa
    *
    * @param tree
    * @param _context
+   * @param context
    */
-  const addExtractersScripts = (tree: Tree, _context: SchematicContext) => {
-    const workspaceProject = getProjectFromTree(tree);
+  const addExtractersScripts = (tree: Tree, context: SchematicContext) => {
+    const workspaceProject = getProjectFromTree(tree, null, 'application');
+
+    if (!workspaceProject) {
+      context.logger.error('No application detected, the extractor scripts will not be added');
```

Comment:
Are you sure we don't want to extract metadata in a library ?
If a project has a library and several applications they would like to extract metadata of the library to be able to specify this library in the field `libraries` in the extractors of their app, no ?